moveit_ros/moveit_servo/src/utils/common.cpp

```diff
@@ -319,27 +387,53 @@ double jointLimitVelocityScalingFactor(const Eigen::VectorXd& velocities,
     min_scaling_factor = 1.0;  // Set to no scaling override.
   }
 
-  // Now get the scaling factor from joint velocity limits.
+  // Now get the scaling factor from joint limits.
   size_t idx = 0;
   for (const auto& joint_bound : joint_bounds)
   {
     for (const auto& variable_bound : *joint_bound)
     {
-      const auto& target_vel = velocities(idx);
-      if (variable_bound.velocity_bounded_ && target_vel != 0.0)
-      {
-        // Find the ratio of clamped velocity to original velocity
-        const auto bounded_vel = std::clamp(target_vel, variable_bound.min_velocity_, variable_bound.max_velocity_);
-        const auto joint_scaling_factor = bounded_vel / target_vel;
-        min_scaling_factor = std::min(min_scaling_factor, joint_scaling_factor);
-      }
+      const auto& joint_target = joint_values(idx);
+      double joint_scaling_factor = calculate_joint_scaling_factor(variable_bound, joint_target);
+      min_scaling_factor = std::min(min_scaling_factor, joint_scaling_factor);
```

Comment:
With velocities it makes sense to slow all of the velocities down by the same ratio. That keeps the robot on the intended path.

It doesn't really make sense with accelerations; it will cause the robot to go off the path. However, Ruckig would handle this well.

Reply:
When you say "path" are you referring to a pre-defined/pre-calculated trajectory? For example, maybe you want to move the robot in a square pattern, so some trajectory optimizer/planner creates the trajectory and then the waypoints are fed into Servo?

Reply:
What i really mean is direction. When this acceleration gets integrated it will change the direction.